src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -91,12 +95,61 @@ public override async Task<IReadOnlyList<PackageSpec>> GetPackageSpecsAsync(Depe
         /// </summary>
         public override async Task<IEnumerable<PackageReference>> GetInstalledPackagesAsync(CancellationToken token)
         {
-            ProjectPackages packages = await GetInstalledAndTransitivePackagesAsync(includeTransitiveOrigins: false, token);
-            return packages.InstalledPackages;
+            token.ThrowIfCancellationRequested();
```

Comment:
Could this have implemented as conditional in `GetInstalledAndTransitivePackagesAsync`?
As current state it's not really large method, I'm not really worried about, but still code duplication.

Reply:
I tried simplifying to one method here https://github.com/NuGet/NuGet.Client/commit/e7d351a2aa230fb304694919627fdab1d7b5ad06

but it has worse performance results if I have both methods. See [comment](https://github.com/NuGet/Client.Engineering/issues/1780#issuecomment-1324299876)

Reply:
Chatted offline, will try this approach by parametrizing `GetInstalledAndTransitivePackagesAsync` and profiling with benchmarkdotnet inspired from [zivkan/Benchmarks](https://github.com/zivkan/Benchmarks) repo sample